enos/modules/vault_unseal_nodes/main.tf

```diff
@@ -0,0 +1,123 @@
+# This module unseals the replication secondary follower nodes
+terraform {
+  required_providers {
+    enos = {
+      source = "app.terraform.io/hashicorp-qti/enos"
+    }
+  }
+}
+
+variable "vault_install_dir" {
+  type        = string
+  description = "The directory where the Vault binary will be installed"
+}
+
+variable "vault_instance_count" {
+  type        = number
+  description = "How many vault instances are in the cluster"
+}
+
+variable "follower_public_ips" {
+  type        = list(string)
+  description = "Vault cluster follower Public IP addresses"
+}
+
+variable "vault_seal_type" {
+  type        = string
+  description = "The Vault seal type"
+}
+
+variable "vault_unseal_keys" {}
+
+locals {
+  followers      = toset([for idx in range(var.vault_instance_count - 1) : tostring(idx)])
+  vault_bin_path = "${var.vault_install_dir}/vault"
+}
+
+# After replication is enabled the secondary follower nodes are expected to be sealed,
+# so we wait for the secondary follower nodes to update the seal status
+resource "enos_remote_exec" "wait_till_sealed" {
+  for_each = {
+    for idx, follower in local.followers : idx => follower
+  }
+  environment = {
+    VAULT_ADDR        = "http://127.0.0.1:8200"
+    vault_install_dir = var.vault_install_dir
+  }
+
+  scripts = ["${path.module}/scripts/wait-till-sealed.sh"]
+
+  transport = {
+    ssh = {
+      host = element(var.follower_public_ips, each.key)
+    }
+  }
+}
+
+# The follower nodes on secondary replication cluster incorrectly report
+# unseal progress 2/3 (Issue: https://hashicorp.atlassian.net/browse/VAULT-12309),
+# so we restart the followers to clear the status and to autounseal incase of awskms seal type
+resource "enos_remote_exec" "restart_followers" {
```

Comment:
I think we should updated VAULT-12309 with acceptance criteria that includes removing this workaround, or creating a task in our backlog that is blocked by VAULT-12309 that removes this.

Reply:
The same feedback applies to the next workarounds and VAULT-12311.

Reply:
Will make a note to add it with links once this PR is merged

Reply:
Just a note here: this behavior (though likely a bug) does not have a huge impact from the end-user's perspective so it is unlikely to get prioritized anytime soon.